cupy/core/core.pyx

```diff
@@ -1644,8 +1644,32 @@ cdef class ndarray:
                 a_gpu = asfortranarray(self)
             else:
                 raise ValueError("unsupported order: {}".format(order))
-        a_cpu = numpy.empty(self._shape, dtype=self.dtype, order=order)
+
+        if out is not None:
+            if not isinstance(out, numpy.ndarray):
+                raise TypeError('Only numpy.ndarray can be obtained from'
+                                'cupy.ndarray')
+            if self.dtype != out.dtype:
+                raise TypeError('{} array cannot be obtained from {} array'.format(
+                    out.dtype, self.dtype))
+            if self.shape != out.shape:
+                raise ValueError(
+                    'Shape mismatch. Expected shape: {}, actual shape: {}'.format(
+                        self.shape, out.shape))
+            if self._c_contiguous:
+                if not out.flags.c_contiguous:
+                    raise RuntimeError('`out` is not c contiguous array')
+            elif self._f_contiguous:
+                if not out.flags.f_contiguous:
+                    raise RuntimeError('`out` is not f contiguous array')
+            else:
+                raise RuntimeError('Cannot set to non-contiguous array')
```

Comment:
This exception message seems wrong. How about this?

```
`out` cannot be specified when copying from non-contiguous ndarray
```

Reply:
Hi,
I think that replace `copying from` with `copying to` in exception message will be more reasonable a bit.
If following code looks OK, I will push commit.
```python
if out.flags.c_contiguous:
  a_gpu = ascontiguousarray(self)
elif out.flags.f_contiguous:
  a_gpu = asfortranarray(self)
else:
  raise RuntimeError('`out` cannot be specified when copying to non-contiguous ndarray')
```